includes/loader.php

```diff
@@ -30,7 +30,16 @@ function autoload_classes() {
 
 	spl_autoload_register(
 		function ( $class ) use ( $class_map ) {
-			if ( isset( $class_map[ $class ] ) ) {
+			if (
+				// Only handle classes defined in our class maps.
+				isset( $class_map[ $class ] )
+				// Only load Site Kit classes or others that exist (e.g. polyfills).
+				&& (
+					0 === strpos( $class, 'Google\\Site_Kit\\' )
+					|| 0 === strpos( $class, 'Google\\Site_Kit_Dependencies\\' )
+					|| file_exists( $class_map[ $class ] )
```

Comment:
I don't think this is a good thing to use the `file_exists` function because it means that for every class autoloaded in other plugins we will check whether that file exists on the disk. Of course will happen only if our autoloader is loaded earlier than autoloaders of other plugins, but anyway generating tons of IO calls for every request is not efficient. I think we need to figure out a better way how we can check if this file is useful and we should require it.

Which non `Google\\Site_Kit*` class do we need to load?

Reply:
> it means that for every class autoloaded in other plugins we will check whether that file exists on the disk

That's not entirely accurate – it would still have to exist in our combined class map here and it would only happen for non-Site Kit classes, so there would only be a handful of classes that it checks.

See https://plugins.svn.wordpress.org/google-site-kit/tags/1.98.0/third-party/vendor/composer/autoload_classmap.php